opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchQueryTest.java

```diff
@@ -0,0 +1,204 @@
+        namedArgument("query", "query_value"),
+        namedArgument("max_expansions", "50")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_prefix_length_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("prefix_length", "0")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_fuzzy_transpositions_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("fuzzy_transpositions", "true")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_fuzzy_rewrite_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("fuzzy_rewrite", "constant_score")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_lenient_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("lenient", "false")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_operator_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("operator", "OR")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_minimum_should_match_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("minimum_should_match", "3")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_zero_terms_query_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("zero_terms_query", "NONE")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_boost_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("boost", "1")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  private class MatchExpression extends FunctionExpression {
```

Comment:
@forestmvey why did you decide to create this class?

Just curious.

Reply:
I created this class the same as done with MatchPhraseQueryTest. This class seemed appropriate for a similar unit test as that with MatchPhraseQueryTest.

Reply:
Right. I see now that I wrote that :/

Probably worth revisiting once they are all in the same branch.